google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/Timestamp.java

```diff
@@ -81,6 +81,10 @@ public static Timestamp ofTimeMicroseconds(long microseconds) {
     long seconds = TimeUnit.MICROSECONDS.toSeconds(microseconds);
     int nanos =
         (int) TimeUnit.MICROSECONDS.toNanos(microseconds - TimeUnit.SECONDS.toMicros(seconds));
+    if (nanos < 0) {
```

Comment:
Maybe a while loop instead?

Reply:
I think this function is much fancier than it has to be. Due to the way Java modulus works, I think
```java
long seconds = microseconds / 1_000_000;
long nanos = microseconds % 1_000_000 * 1000;
```
should just work.

Reply:
@pongad `nanos` might still be negative this way. We would still need to make it positive by taking 1 second from `seconds`.
@garrettjonesgoogle  `nanos` wouldn't be less than `1_000_000_000` since it only contains the fraction less than 1 second. So an `if` should be enough.